lib/core_extensions.rb

```diff
@@ -38,10 +38,20 @@ def initialize(*attribute)
     end
 
     def before_destroy(record)
-      klasses.each do |klass|
-        record.send(klass.to_sym).each do |what|
-          what = what.to_label unless what.is_a? String
-          record.errors.add :base, _("%{record} is used by %{what}") % { :record => record, :what => what }
+      klasses.each do |klass, klass_name = klass|
+        record.association(klass.to_sym).association_scope.each do |what|
+          error_message = _("%{record} is used by %{what}")
+          unless what.is_a? String
+            authorized_associations = AssociationAuthorizer.authorized_associations(record.class.reflect_on_association(klass.to_sym).klass, klass_name, false)
+            if (!authorized_associations.respond_to?(:to_a) || authorized_associations.to_a.include?(what)) &&
+                what.class.find_by_id(what.id)
```

Comment:
In was the original solution for different org/loc but it looks like authorized_associations already handles it, I'll remove it.